doozerlib/cli/scan_sources.py

```diff
@@ -47,37 +47,37 @@ def config_scan_source_changes(runtime, ci_kubeconfig, as_yaml):
     changing_rpm_packages = set()
     assessment_reason = dict()  # maps metadata qualified_key => message describing change
 
-    def add_assessment_reason(meta, changing, reason):
+    def add_assessment_reason(meta, rebuild_hint: RebuildHint):
         # qualify by whether this is a True or False for change so that we can store both in the map.
-        key = f'{meta.qualified_key}+{changing}'
+        key = f'{meta.qualified_key}+{rebuild_hint.rebuild}'
         # If the key is already there, don't replace the message as it is likely more interesting
         # than subsequent reasons (e.g. changing because of ancestry)
         if key not in assessment_reason:
-            assessment_reason[key] = reason
+            assessment_reason[key] = rebuild_hint.reason
 
-    def add_image_meta_change(meta, msg):
+    def add_image_meta_change(meta, rebuild_hint: RebuildHint):
         nonlocal changing_image_metas
         changing_image_metas.add(meta)
-        add_assessment_reason(meta, True, msg)
+        add_assessment_reason(meta, rebuild_hint)
```

Comment:
This function doesn't check `rebuild_hint.rebuild`. Would like to add `assert rebuild_hint.rebuild is True` here in case we forget.